### PR TITLE
ODC-6023-update devconsole-ci-tests

### PR DESCRIPTION
**Pull request by makambalaji**
**Jira Id**: https://issues.redhat.com/browse/ODC-6023

**Description**:
CI implemented for entire add flow

**Browser & its version** : Chrome 91

**Pre-conditions/setup:**

**Checks for approving Epic scenarios Automation PR:**

- [ ] Execute the @to-do tagged gherkin scripts manually
- [ ] Convert the @to-do gherkin scripts to cypress automation scripts
- [ ] Once scripts are automated, replace tag @to-do with @epic-number
- [ ] Execute the scripts in Remote cluster

**Refactoring criteria for approving Automation PR:**

- [ ] update the test cases count in [Automation status confluence Report](https://docs.jboss.org/display/ODC/Automation+Status+Report)

**Execution Commands**:
In below commands, please update url and password based on the cluster
```
export NO_HEADLESS=true && export CHROME_VERSION=$(/usr/bin/google-chrome-stable --version)
BRIDGE_KUBEADMIN_PASSWORD=YH3jN-PRFT2-Q429c-5KQDr
BRIDGE_BASE_ADDRESS=https://console-openshift-console.apps.dev-svc-4.8-042801.devcluster.openshift.com
export BRIDGE_KUBEADMIN_PASSWORD
export BRIDGE_BASE_ADDRESS
oc login -u kubeadmin -p $BRIDGE_KUBEADMIN_PASSWORD https://api.gamore48installer7thjune001.devcluster.openshift.com:6443  --insecure-skip-tls-verify
oc apply -f ./frontend/integration-tests/data/htpasswd-secret.yaml
oc patch oauths cluster --patch "$(cat ./frontend/integration-tests/data/patch-htpasswd.yaml)" --type=merge
./test-cypress.sh -p dev-console -h true
```
**Results**: 18 test cases are executed in 19 min [duration]
**Screenshots**
image

**Browser conformance**:

- [x] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

### CODE REVIEW

File: frontend/packages/dev-console/integration-tests/package.json
Comment (jerolimov) [inline frontend/packages/dev-console/integration-tests/package.json:14]: Should this `package.json` script test all features (smoke tests)??

**Review by jerolimov — APPROVED** [review]
Verified this with a local run of cypress via `yarn test-cypress-devconsole`

With this PR:
with-pr

Before:
master

Error details on master branch:
error

/lgtm
/retest

File: frontend/packages/dev-console/integration-tests/package.json
Comment (makambalaji) [reply frontend/packages/dev-console/integration-tests/package.json:14]: Yes @jerolimov, it executes all Add flow test cases based on the tags mentioned in [cypress.json](https://github.com/openshift/console/blob/master/frontend/packages/dev-console/integration-tests/cypress.json#L22)

File: frontend/packages/dev-console/integration-tests/support/pageObjects/add-flow-po.ts
Comment (jerolimov) [inline frontend/packages/dev-console/integration-tests/support/pageObjects/add-flow-po.ts:116]: If you like you can also add the following data-tests attributes and change this here to this new data-test references. Wdyt?

```suggestion
    helmCharts: '[data-test="tab HelmChart"]',
    builderImage: '[data-test="tab BuilderImage"]',
    template: '[data-test="tab Template"]',
```

File: frontend/packages/dev-console/integration-tests/support/pageObjects/add-flow-po.ts
Comment (jerolimov) [inline frontend/packages/dev-console/integration-tests/support/pageObjects/add-flow-po.ts:119]:
```suggestion
    eventSources: '[data-test="tab EventSource"]',
```

**Review by jerolimov — COMMENTED** [review]
If you like you can also add the following data-tests attributes and change this here to this new data-test references. Wdyt?

* In `CatalogCategories.tsx` I would recommend:

```diff
    return (
      <VerticalTabsTab
        key={id}
        title={label}
        active={active}
        className={tabClasses}
        onActivate={() => onSelectCategory(id)}
        hasActiveDescendant={hasActiveDescendant(selectedCategory, category)}
        shown={toplevelCategory}
+        data-test={`tab ${id}`}
      >
        {subcategories && (
          <VerticalTabs restrictTabs activeTab={isActiveTab(selectedCategoryID, category)}>
            {_.map(subcategories, (subcategory) =>
              renderTabs(subcategory, selectedCategoryID, false),
            )}
          </VerticalTabs>
        )}
      </VerticalTabsTab>
    );
  };

  return (
-    <VerticalTabs restrictTabs activeTab={activeTab}>
+    <VerticalTabs restrictTabs activeTab={activeTab} data-test="catalog-categories">
      {_.map(categories, (category) => renderTabs(category, selectedCategory, true))}
    </VerticalTabs>
  );
```

* In `CatalogTypeSelector.tsx`:

```diff
-      <VerticalTabs>
+      <VerticalTabs data-test="catalog-types">
        {catalogTypes.map((type) => {
          const { value, label } = type;
          const typeCount = catalogTypeCounts[value];
          const queryParams = new URLSearchParams(search);
          queryParams.set(CatalogQueryParams.TYPE, type.value);

          const to = {
            pathname,
            search: `?${queryParams.toString()}`,
          };

          return typeCount > 0 ? (
-            <li key={value} className="vertical-tabs-pf-tab">
+            <li key={value} className="vertical-tabs-pf-tab" data-test={`tab ${value}`}>
              <Link to={to}>{`${label} (${typeCount})`}</Link>
            </li>
          ) : null;
        })}
      </VerticalTabs>
```

Wdyt?

File: frontend/packages/dev-console/integration-tests/package.json
Comment (jerolimov) [reply frontend/packages/dev-console/integration-tests/package.json:14]: Buy why? Why not all? :)

**Review by jerolimov — APPROVED** [review]
Run the cypress tests locally again, and run successfully. Selectors looks good.
/lgtm

Approve the `test-data` frontend changes.
/approve

/retest

**Comment by jerolimov** [comment]
/hold
Looks like CI a11y checks are broken. I will take a look.

**Comment by jerolimov** [comment]
The E2E tests fails because of an OperatorHub I fixed now in #9479. We need to wait for this PR and after that this PR should be rebased and tested after that again.

@makambalaji As mentioned in slack, I still got an issue with multiple entries in the `OperatorHub`. As we discussed it might be good to use full qualified data-test ids here

https://github.com/openshift/console/blob/master/frontend/packages/dev-console/integration-tests/support/pageObjects/operators-po.ts#L16-L25

File: frontend/packages/dev-console/integration-tests/package.json
Comment (makambalaji) [reply frontend/packages/dev-console/integration-tests/package.json:14]: If we need to execute all, it takes more time.. which is not required for CI

**Comment by makambalaji** [comment]
> The E2E tests fails because of an OperatorHub I fixed now in #9479. We need to wait for this PR and after that this PR should be rebased and tested after that again.
> 
> @makambalaji As mentioned in slack, I still got an issue with multiple entries in the `OperatorHub`. As we discussed it might be good to use full qualified data-test ids here
> 
> https://github.com/openshift/console/blob/master/frontend/packages/dev-console/integration-tests/support/pageObjects/operators-po.ts#L16-L25

updated the page objects

**Comment by jerolimov** [comment]
Tested this PR together with this two PRs. After that the headless tests run successfully on a shared cluster and a local cluster (crc). Sorry but I think we should wait until both are merged and then this PR should be rebased again.

* #9499
* #9500

```
cd packages/dev-console/integration-tests
yarn test-cypress-headless
```

**Comment by makambalaji** [comment]
> Tested this PR together with this two PRs. After that the headless tests run successfully on a shared cluster and a local cluster (crc). Sorry but I think we should wait until both are merged and then this PR should be rebased again.
> 
> * [Fix a11y form errors when import a container image #9499](https://github.com/openshift/console/pull/9499)
> * [Fix a11y file input form issue, for example when creating a Secret #9500](https://github.com/openshift/console/pull/9500)
> 
> ```
> cd packages/dev-console/integration-tests
> yarn test-cypress-headless
> ```

Looks like pr's are merged

**Comment by makambalaji** [comment]
data-test attributes are included in this pr. due to that reason, scripts are verified on localhost instead of remote cluster once this pr is merged. will raise a separate pr for modifying devconsole/integration-tests/package.json to execute all addFlow scripts

**Comment by jerolimov** [comment]
Re run the tests locally, and works fine.

/lgtm

**Pull request merged**
